subprojects/files/src/main/java/org/gradle/api/internal/file/archive/impl/FileZipInput.java

```diff
@@ -52,16 +52,16 @@ public static ZipInput create(File file) {
      * {@link ZipFile} is more efficient, but causes memory leaks on older Java versions, so we only use it on more recent ones.
      */
     private static boolean isZipFileSafeToUse() {
-        return getJavaMajorVersion() >= 11;
+        return isJava11OrLater();
     }
 
-    private static int getJavaMajorVersion() {
-        String versionString = System.getProperty("java.version");
+    private static boolean isJava11OrLater() {
```

Comment:
I think it should be fine to inline this method, especially since the explanation for the whole Java 11 thing is described in the Javadoc for the caller method.

I'd place a comment here describing why we still need to split at `.`, i.e. because `java.specification.version` looks like `1.8` up until Java 8, and then `10` etc. later.